### PR TITLE
MINOR: Rename RecordBatch to ProducerBatch to allow reuse in KIP-98

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by hachikuji** [comment]
cc @ijuma, as discussed in #2614.

**Comment by apurvam** [comment]
Looks like this needs a rebase?

**Review by apurvam — APPROVED** [review]
LGTM!

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, looks good overall. We still have a things like `IncompleteRecordBatches` and `RecordBatchTooLargeException`. The latter is probably OK to stay, but what about the former?

File: clients/src/test/java/org/apache/kafka/clients/producer/internals/ProducerBatchTest.java [hunk 1]
```diff
@@ -16,56 +16,57 @@
...
         batch.reenqueued(now);
         // Set `now` to 2ms before the create time.
         assertFalse(batch.maybeExpire(10240, Long.MAX_VALUE, now - 2L, 10240L, false));
     }
 
     /**
-     * A RecordBatch#maybeExpire call with a now value before the create time of the RecordBatch is correctly recognized
-     * as not expired when invoked with parameter isFull = true.
+     * A {@link ProducerBatch#maybeExpire(int, long, long, long, boolean)} call with a now value before the create
+     * time of the RecordBatch is correctly recognized as not expired when invoked with parameter isFull = true.
```
Comment (ijuma) [inline clients/src/test/java/org/apache/kafka/clients/producer/internals/ProducerBatchTest.java]: RecordBatch -> ProducerBatch

**Comment by hachikuji** [comment]
@ijuma Thanks for reviewing. Agreed on `IncompleteRecordBatches`. I will rename it as well.

**Review by ijuma — APPROVED** [review]
Thanks, LGTM. Merging to trunk.

**Pull request closed without merging**